sdktests/custom_matchers_events.go

```diff
@@ -0,0 +1,93 @@
+package sdktests
+
+import (
+	m "github.com/launchdarkly/go-test-helpers/v2/matchers"
+	"gopkg.in/launchdarkly/go-sdk-common.v2/lduser"
+)
+
+// These are used with the matchers API to make assertions about JSON event data. The value
+// representing an individual analytics event uses the type mockld.Event, which we treat here
+// as an opaque wrapper for a JSON object. The matchers can use the transformation m.JSONMap()
+// to treat that object as a map, and then make assertions about the key and values in the map;
+// or, for simpler assertions about a single property, they use the shortcut m.JSONProperty().
+//
+// In order to make tests as granular and useful as possible, we don't want to always be
+// making equality assertions about entire events at once; if we did, then an SDK bug that
+// affects just one detail of an event would cause failures in many tests that were supposed
+// to be about other things, making it harder to isolate the problem. Therefore, we have very
+// basic matchers like IsIdentifyEvent which just verify the kind of the event, and also ones
+// like IsIdentifyEventForUserKey that look at an additional property to make sure it is for
+// the right user, without verifying all of the properties.
+
+func EventHasKind(kind string) m.Matcher {
+	return m.JSONProperty("kind").Should(m.Equal(kind))
+}
+
+func HasContextKind(user lduser.User) m.Matcher {
+	if user.GetAnonymous() {
+		return m.JSONProperty("contextKind").Should(m.Equal("anonymousUser"))
+	}
+	return JSONPropertyNullOrAbsent("contextKind")
+}
+
+func HasAnyCreationDate() m.Matcher {
+	return m.JSONProperty("creationDate").Should(ValueIsPositiveNonZeroInteger())
```

Comment:
If a value is positive, it is already defined as non zero so you could theoretically simply this name some to ValueIsPositiveInteger. If you wanted a positive or zero value, then we could have ValueIsNonNegativeInteger. But maybe that's too pedantic?

Reply:
You're strictly correct, but non-mathematicians so often conflate "positive" with "nonnegative" that I prefer to spell it out like this for clarity.